flutter_local_notifications/android/src/main/java/com/dexterous/flutterlocalnotifications/FlutterLocalNotificationsPlugin.java

```diff
@@ -282,7 +282,15 @@ private static void scheduleNotification(Context context, final NotificationDeta
         }
     }
 
-    private static void zonedScheduleNotification(Context context, final NotificationDetails notificationDetails, Boolean updateScheduledNotificationsCache) {
+    private static void scheduleNotificationsAndUpdateCache(Context context, final List<NotificationDetails> notificationDetails) {
+        for (NotificationDetails details : notificationDetails) {
```

Comment:
IMO, saving should happen after a scheduling a single notification so it's treated like an atomic transaction. For example, if scheduling was to fail halfway through the collection for some reason, the ones that were successfully scheduled should have been saved to shared preferences

Reply:
good point. however, the saving was one of the main time-consumers, and iirc it can take up to 50ms. 
a solution could be to add each successfully scheduled notification to a temporary list, and then save this in one go.